### ISSUE TITLE
Extended configuration does not apply to JS files

### ISSUE THREAD

**Issue by deskoh** [palantir/tslint#4564]
### Feature request

Currently using using a tslint rules that extends `tslint-config-airbnb`. However, the rules does not seem to apply to JS files after setting `"jsRules"` property set to `true`. The [documentation](https://palantir.github.io/tslint/usage/configuration/) also did not specify whether this is the intended behaviour.

Example `tslint.json` file:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint-config-airbnb"
  ],
  "jsRules": true
}
```

```JavaScript
function test() {
  // No lint errors for `prefer-const` rule even though it is defined in `tslint-config-airbnb`.
  let x = 1;
}
```

The `prefer-const` rule need to be added explicitly:
```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint-config-airbnb"
  ],
  "jsRules": {
    "prefer-const": true,
  }
}
```

Any solution for the extended rules to also apply to JS files?

**Comment by JoshuaKGoldberg** [comment]
👋 @deskoh, is this a duplicate of #4513?

**Comment by deskoh** [comment]
Slightly different. Currently using `tsc 5.13.1`.

`tsconfig.json` has `"allowJs"` set to `true`, running `tslint -p .` for the following `tslint.json` has different results:

```JSON
// Does not flag prefer-const errors in JS files
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": true,
    "rules": {},
    "rulesDirectory": []
}
```

```JSON
// Flag prefer-const errors in JS files
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {
        "prefer-const": true
    },
    "rules": {},
    "rulesDirectory": []
}
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
